test/test_settings.py

```diff
@@ -367,125 +369,65 @@ class TestSettingsEquality(object):
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=True, initial_values=overrides)
-
-    def another_instance(self):
-        """
-        Return an instance of the class under test.  Each call to this method
-        must return a different object.  The objects must not be equal to the
-        objects returned by an_instance.  They may or may not be equal to
-        each other (they will not be compared against each other).
-        """
-        overrides = {
-            h2.settings.SettingCodes.HEADER_TABLE_SIZE: 8080,
-            h2.settings.SettingCodes.MAX_FRAME_SIZE: 16388,
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 100,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=False, initial_values=overrides)
-
-    def test_identical_eq(self):
-        """
-        An object compares equal to itself using the == operator.
-        """
-        o = self.an_instance()
-        assert (o == o)
-
-    def test_identical_ne(self):
-        """
-        An object doesn't compare not equal to itself using the != operator.
-        """
-        o = self.an_instance()
-        assert not (o != o)
-
-    def test_same_eq(self):
-        """
-        Two objects that are equal to each other compare equal to each other
-        using the == operator.
-        """
-        a = self.an_instance()
-        b = self.an_instance()
-        assert (a == b)
-
-    def test_same_ne(self):
-        """
-        Two objects that are equal to each other do not compare not equal to
-        each other using the != operator.
-        """
-        a = self.an_instance()
-        b = self.an_instance()
-        assert not (a != b)
+    SettingsStrategy = builds(
+        h2.settings.Settings,
+        client=booleans(),
+        initial_values=fixed_dictionaries({
+            h2.settings.SettingCodes.HEADER_TABLE_SIZE: integers(0),
+            h2.settings.SettingCodes.ENABLE_PUSH: integers(0, 1),
+            h2.settings.SettingCodes.INITIAL_WINDOW_SIZE:
+                integers(0, 2**31 - 1),
+            h2.settings.SettingCodes.MAX_FRAME_SIZE: integers(16384, 16777215),
+            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: integers(0),
+            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: integers(0),
+        })
+    )
 
-    def test_different_eq(self):
+    @given(settings=SettingsStrategy)
+    def test_equality_reflexive(self, settings):
         """
-        Two objects that are not equal to each other do not compare equal to
-        each other using the == operator.
+        An object compares equal to itself using the == operator and the !=
+        operator.
         """
-        a = self.an_instance()
-        b = self.another_instance()
-        assert not (a == b)
+        assert (settings == settings)
+        assert not (settings != settings)
```

Comment:
I’ve realised since writing my last comment that this isn’t always the best test, because you’re comparing the identity of two identical objects. Python will default to using identity as an equality test, so this passes even if you don't supply custom equality methods:

```pycon
>>> class Foo: pass
...
>>> x = Foo()
>>> y = Foo()
>>> x == x
True
>>> x == y
False
>>> x != x
False
```

My preferred approach is now to test based on copies of objects, i.e.,

```python
other_settings = copy.deepcopy(settings)
assert (settings == other_settings)
assert not (settings != other_settings)
```

although this does rely on you being able to copy your objects, and I can’t remember what the options are here for `IntEnum` and its subclasses.